src/api/schema/components/mod.rs

```diff
@@ -45,31 +47,37 @@ impl Component {
         }
     }
 
-    fn get_filter_type(&self) -> ComponentType {
+    fn get_component_kind(&self) -> ComponentKind {
         match self {
-            Component::Source(_) => ComponentType::Source,
-            Component::Transform(_) => ComponentType::Transform,
-            Component::Sink(_) => ComponentType::Sink,
+            Component::Source(_) => ComponentKind::Source,
+            Component::Transform(_) => ComponentKind::Transform,
+            Component::Sink(_) => ComponentKind::Sink,
         }
     }
 }
 
 #[derive(Default, InputObject)]
 pub struct ComponentsFilter {
     name: Option<Vec<filter::StringFilter>>,
-    component_type: Option<Vec<filter::EqualityFilter<ComponentType>>>,
+    component_kind: Option<Vec<filter::EqualityFilter<ComponentKind>>>,
     or: Option<Vec<Self>>,
 }
 
+#[derive(Enum, Copy, Clone, Eq, PartialEq)]
+pub enum ComponentsSortFieldName {
```

Comment:
A field is typically an enum in GraphQL, for type safety (although any `async_graphql::InputType` will work.)